tests/integration-tests/tests/performance_tests/test_starccm.py

```diff
@@ -0,0 +1,72 @@
+import pytest
+import logging
+import boto3
+import json
+
+from remote_command_executor import RemoteCommandExecutor, RemoteCommandExecutionError
+
+# timeout in seconds
+INSTALLER_TIMEOUT = 1800
+SCHEDULER_TIMEOUT = 600
+STARCCM_SECRET_NAME = 'perf_integ_secret'
+NODES = [8, 16, 32]
+
+# TODO: replace with 8,16,32 nodes, test with 2 compute nodes, injected into pcluster config reader
+
+
+def get_starccm_secrets(region_name):
+    secrets_manager_client = boto3.client('secretsmanager', region_name=region_name)
+    response = secrets_manager_client.get_secret_value(SecretId="arn:aws:secretsmanager:eu-west-1:815618621112:secret:perf_integ_secret-iZqSUK")["SecretString"]
+    secrets = json.loads(response)
+    return secrets["podkey"], secrets["licpath"]
+
+
+def starccm_installed(headnode):
+    cmd = "/shared/ec2-user/STAR-CCM+/16.02.008/STAR-CCM+16.02.008/star/bin/starccm+ --version"
+    try:
+        headnode.run_remote_command(cmd)
+        return True
+    except RemoteCommandExecutionError:
+        return False
+
+@pytest.mark.parametrize(
+    "number_of_nodes",
+    [
+        32,
+    ],
+)
+def test_starccm(
+        vpc_stack,
+        instance,
+        os,
+        region,
+        scheduler,
+        pcluster_config_reader,
+        clusters_factory,
+        number_of_nodes,
+        test_datadir,
+        scheduler_commands_factory,
+):
+    cluster_config = pcluster_config_reader(number_of_nodes=number_of_nodes)
+    cluster = clusters_factory(cluster_config)
+    logging.info("Cluster Created")
+    remote_command_executor = RemoteCommandExecutor(cluster)
+    if not starccm_installed(remote_command_executor):
+        logging.info("Installing StarCCM+")
+        remote_command_executor.run_remote_script(str(test_datadir / "starccm.install.sh"),
+                                                  timeout=INSTALLER_TIMEOUT, hide=False)
+    logging.info("StarCCM+ Installed")
+    scheduler_commands = scheduler_commands_factory(remote_command_executor)
+    measurements = []
+    secret1, secret2 = get_starccm_secrets(region)
+
+    for node in NODES:
```

Comment:
You're iterating here on an array that is not derived from the number_of_nodes. This is not a robust solution because if I set the number_of_nodes to 16 I also need to change the NODES constant accordingly. We can consider this fine for the first iteration but we need to improve the solution in the next iter